lib/remote.provider.js

```diff
@@ -1,7 +1,148 @@
+      APP_NAME,
+      null,
+      INTERFACES,
+      options,
+      (error, connection, api) => {
+        if (error) {
+          callback(error);
+          return;
+        } else {
+          this.api = api;
+          callback(null, this);
+        }
+      }
+    );
+  }
+
+  // Close RemoteProvider
+  //   callback - function(err)
+  // eslint-disable-next-line no-unused-vars
+  close(callback) {
+    if (!this.api) {
+      callback();
+      return;
+    }
+    this.api = null;
+  }
+
+  // Generate globally unique id
+  //   callback - function(err, id)
+  // eslint-disable-next-line no-unused-vars
+  takeId(callback) {
+    throw new GSError(errorCodes.NOT_IMPLEMENTED);
+  }
+
+  // Get record from GlobalStorage
+  //   id - globally unique record id
+  //   callback - function(err, obj)
+  // eslint-disable-next-line no-unused-vars
+  get(id, callback) {
+    this.api.provider.get(id, (error, callback));
+
+    getRecordMethod(id).then(record => callback(record));
+  }
+
+  // Set record in GlobalStorage
+  //   record - record to be stored
+  //   callback - function(err)
+  // eslint-disable-next-line no-unused-vars
+  set(record, callback) {
+    const setRecordMethod = createMethod(
+      api => api.provider.set,
+      this.apiPromise
+    );
+
+    setRecordMethod(record).then(() => callback());
+  }
+
+  // Create record in GlobalStorage
+  //   record - record to be stored
+  //   callback - function(err, id)
+  // eslint-disable-next-line no-unused-vars
+  create(category, record, callback) {
+    this.api.provider.create(
+      category,
+      record,
+      (error,
+      id => {
+        if (error) {
+          callback(error);
+          return;
+        }
+        callback(null, id);
+      })
+    );
+  }
+
+  // Update record in GlobalStorage
+  //   query - record, example: { Id }
+  //   patch - record, fields to update
+  //   callback - function(err, count)
+  // eslint-disable-next-line no-unused-vars
+  update(category, query, patch, callback) {
+    this.api.provider.update(category, query, patch, (error, count) => {
+      if (error) {
+        callback(error);
+        return;
+      }
+      callback(null, count);
+    });
+  }
+
+  // Delete record in GlobalStorage
+  //   query - record, example: { Id }
+  //   callback - function(err, count)
+  // eslint-disable-next-line no-unused-vars
+  delete(category, query, callback) {
+    this.api.provider.delete(category, query, (error, count) => {
+      if (error) {
+        callback(error);
+        return;
+      }
+      callback(null, count);
+    });
+  }
+
+  // Select record from GlobalStorage
+  //   query - fields conditions
+  // Returns: Cursor
+  // eslint-disable-next-line no-unused-vars
+  select(category, query, callback) {
+    this.api.provider.select(category, query, (error, cursorId) => {
+      if (error) {
+        callback(error);
+        return;
+      }
+      const cursor = new RemoteCursor(cursorId);
```

Comment:
Yous should also pass `api` or `connection` to cursor, BTW where is RemoteCursor implementation?

Reply:
@nechaido, Remote Cursor is another issue #214, and PR - do not apply to this task